pkg/asset/agent/image/oc.go

```diff
@@ -145,12 +160,93 @@ func (r *release) extractFileFromImage(log logrus.FieldLogger, image, file, cach
 	}
 	// set path
 	path := filepath.Join(cacheDir, path.Base(file))
-	log.Info("Successfully extracted base ISO from the release")
-	log.Debugf("Base ISO %s cached at %s", file, path)
 	return path, nil
 }
 
-func execute(log logrus.FieldLogger, executer executer.Executer, pullSecret string, command string) (string, error) {
+// Get hash from rhcos.json
+func getHashFromInstaller(architecture string) (bool, string) {
+
+	// Get hash from metadata in the installer
+	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
+	defer cancel()
+
+	st, err := rhcos.FetchCoreOSBuild(ctx)
+	if err != nil {
+		return false, ""
+	}
+
+	streamArch, err := st.GetArchitecture(architecture)
+	if err != nil {
+		return false, ""
+	}
+	if artifacts, ok := streamArch.Artifacts["metal"]; ok {
+		if format, ok := artifacts.Formats["iso"]; ok {
+			return true, format.Disk.Sha256
+		}
+	}
+
+	return false, ""
+}
+
+func matchingHash(imageSha []byte, sha string) bool {
+	decoded, err := hex.DecodeString(sha)
+	if err == nil && reflect.DeepEqual(imageSha, decoded) {
```

Comment:
nit: could use `bytes.Equal()` here rather than resorting to the reflect module.

Reply:
Done